src/Symfony/Component/Translation/Loader/XliffFileLoader.php

```diff
@@ -69,12 +108,29 @@ public function load($resource, $locale, $domain = 'messages')
 
             $catalogue->setMetadata((string) $source, $metadata, $domain);
         }
+    }
 
-        if (class_exists('Symfony\Component\Config\Resource\FileResource')) {
-            $catalogue->addResource(new FileResource($resource));
-        }
+    /**
+     * @param \DOMDocument     $dom
+     * @param MessageCatalogue $catalogue
+     * @param string           $domain
+     */
+    private function extractXliff2(\DOMDocument $dom, MessageCatalogue $catalogue, $domain)
+    {
+        $xml = simplexml_import_dom($dom);
+        $encoding = strtoupper($dom->encoding);
 
-        return $catalogue;
+        $xml->registerXPathNamespace('xliff', 'urn:oasis:names:tc:xliff:document:2.0');
+
+        foreach ($xml->xpath('//xliff:unit/xliff:segment') as $segment) {
+            $source = $segment->source;
+
+            // If the xlf file has another encoding specified, try to convert it because
+            // simple_xml will always return utf-8 encoded values
+            $target = $this->utf8ToCharset((string) (isset($translation->target) ? $translation->target : $source), $encoding);
+
+            $catalogue->set((string) $source, $target, $domain);
```

Comment:
I only added support for target attributes
but for notes there's no way to do a mapping because aren't assigned to specific segments, but to whole units.